### PR TITLE
chore: name refactor and enabling migration 012 by adding changeunit annotation

### PR DESCRIPTION
**Pull request by sondermanish**
## Description

This PR just uncomments the migration annotation in order to make this migration effective, This annotation was commented to implement the EE override before execution of this migration file.

Note: This PR needs to be merged only when we have placed a merge freeze 
This should be merged before the merge of PR: https://github.com/appsmithorg/appsmith-ee/pull/1533

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

#### How Has This Been Tested?
- [ ] Manual

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### CODE REVIEW

**Review by sumitsum — DISMISSED** [review]

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration109TransferToDatasourceStorage.java [hunk 1]
```diff
@@ -27,48 +28,37 @@
...
     public void rollbackExecution() {
         // We're handling rollbacks using marker fields, so we don't need to implement this
     }
 
-    //    @Execution
+    @Execution
     public void executeMigration() {
-        // First fetch all datasource ids and workspace ids for datasources that
-        // do not have `hasDatasourceStorage` value set as true
-        final Query datasourceQuery = query(findDatasourceIdsToUpdate()).cursorBatchSize(1024);
```
Comment (sumitsum) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration109TransferToDatasourceStorage.java:51]: QQ: Why is this section removed now ?

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration109TransferToDatasourceStorage.java [hunk 2]
```diff
@@ -27,48 +28,37 @@
...
-//@ChangeUnit(order = "012", id = "migrate-configurations-to-data-storage", author = " ")
-public class Migration012TransferToDatasourceStorage {
+@ChangeUnit(order = "109", id = "migrate-configurations-to-data-storage", author = " ")
+// Although this migration is common to CE and BE, the ordering of this change-unit needs to be specifically after 107-ee,
+// as this migration has EE overrides for default environmentId, and for getting the environments on existing workspaces, 107-ee needs to run
+public class Migration109TransferToDatasourceStorage {
     private final MongoTemplate mongoTemplate;
 
     private final String migrationFlag = "hasDatasourceStorage";
+    private static final String datasourceConfigurationConstant = fieldName(QDatasource.datasource.datasourceConfiguration);
```
Comment (sumitsum) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration109TransferToDatasourceStorage.java:38]: Minor: Hi Manish, Can you please rename this to `datasourceConfigurationFieldName` ? Similarly for the next variable too.

**Review by sumitsum — CHANGES_REQUESTED** [review]

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration109TransferToDatasourceStorage.java (hunk 1 above)
Comment (sondermanish) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/db/ce/Migration109TransferToDatasourceStorage.java:51]: We were creating a map of datasourceId to workspaceId, which isn't required as we are streaming through the datasources itself to migrate it, we can do without that db call

**Review by sumitsum — APPROVED** [review]

**Pull request merged**